enumerables.rb

```diff
@@ -0,0 +1,172 @@
+      my_each { |item| outcome = false unless yield(item) }
+      outcome
+    elsif arg.is_a?(Class)
+      outcome = true
+      my_each { |item| outcome = false unless item.is_a?(arg) }
+      outcome
+    elsif arg.is_a?(Integer)
+      outcome = true
+      my_each { |item| outcome = false if item != arg }
+      outcome
+    elsif arg.is_a?(String) || arg.is_a?(Regexp)
+      outcome = true
+      my_each { |item| outcome = false unless item.match?(arg) }
+      outcome
+    end
+  end
+
+  def my_any?(arg = nil)
+    if arg.nil? && !block_given?
+      output = true
+      my_each { |item| output = false if !item || item.nil? }
+      output
+    elsif block_given?
+      outcome = false
+      my_each { |item| outcome = true if yield(item) }
+      outcome
+    elsif arg.is_a?(Class)
+      outcome = false
+      my_each { |item| outcome = true if item.is_a?(arg) }
+      outcome
+    elsif arg.is_a?(Integer)
+      outcome = false
+      my_each { |item| outcome = true if item == arg }
+      outcome
+    elsif arg.is_a?(String) || arg.is_a?(Regexp)
+      outcome = false
+      my_each { |item| outcome = true if item.match?(arg) }
+      outcome
+
+    end
+  end
+
+  def my_none?(arg = nil)
+    if arg.nil? && !block_given?
+      output = true
+      my_each { |item| output = false if !item || item.nil? }
+      output
+    elsif block_given?
+      outcome = true
+      my_each { |item| outcome = false if yield(item) }
+      outcome
+    elsif arg.is_a?(Class)
+      outcome = true
+      my_each { |item| outcome = false if item.is_a?(arg) }
+      outcome
+    elsif arg.is_a?(Integer)
+      outcome = true
+      my_each { |item| outcome = false if item == arg }
+      outcome
+    elsif arg.is_a?(String) || arg.is_a?(Regexp)
+      outcome = true
+      my_each { |item| outcome = false if item.match?(arg) }
+      outcome
+    end
+  end
+
+  def my_count(arg = nil)
+    n = 0
+    if block_given?
+      my_each { |item| n += 1 if yield(item) }
+    elsif arg.nil?
+      n += 1 while n < size
+    elsif arg.is_a?(Class)
+      my_each { |item| n += 1 if item.is_a?(arg) }
+    elsif arg.is_a?(Integer)
+      my_each { |item| n += 1 if item == arg }
+    else arg.is_a?(String)
+         my_each { |item| n += 1 if item == arg || item.match?(arg) }
+    end
+    n
+  end
+
+  def my_map(arg = nil)
+    if arg.nil?
+      return to_enum unless block_given?
+
+      output = []
+      my_each { |item| output.push(yield(item)) }
+      output
+    elsif arg.class == Proc
+      my_each { |item| Proc.call(item) }
+    elsif arg.is_a?(Class)
+      output = []
+      my_each { |item| output.push(item) if item.is_a?(arg) }
+      output
+    elsif arg.is_a?(Integer)
+      output = []
+      my_each { |item| output.push(item) if item == arg }
+      output
+    else arg.is_a?(String)
+         output = []
+         my_each { |item| output.push(item) if item == arg || item.match?(arg) }
+         output
+
+    end
+    output
+  end
```

Comment:
```rb
2.6.5 :018 > my_proc = proc {|num| num > 10 }
2.6.5 :019 > array.my_map(my_proc) {|num| num < 10 }
Traceback (most recent call last):
       11: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `<main>'
       10: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `eval'
        9: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `<main>'
        8: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `load'
        7: from /home/maelfosso/.rvm/gems/ruby-2.6.5/gems/irb-1.2.4/exe/irb:11:in `<top (required)>'
        6: from (irb):19
        5: from enumerables.rb:126:in `my_map'
        4: from enumerables.rb:6:in `my_each'
        3: from enumerables.rb:6:in `times'
        2: from enumerables.rb:7:in `block in my_each'
        1: from enumerables.rb:126:in `block in my_map'
NoMethodError (undefined method `call' for Proc:Class)
2.6.5 :020 > 
```

Please, keep in mind that When both a block and a proc are given and it returns a new array with the results of running the proc once for every element in array.

Reply:
regarding this issue, 
you said that my_map must not work unless a block and a proc are given together and when running it with my_map it returns an error but u didn't try to run with the regular map method because it returns the same error as well.
also, the point of having procs is to predefine a block so that u don't write it over and over again which doesn't make sense when you predefine a block with a proc and then try to write it again with the proc...